### ISSUE TITLE
Support for 64-bit

### ISSUE THREAD

**Issue by etruta** [ewindso/ios-parse-titanium-module#31]
Starting February 1st 2015 Apple will require that all App Store submissions be compiled using the iOS 8 SDK, and support 64-bit architectures.

http://www.appcelerator.com/blog/2014/11/alpha-version-of-titanium-sdk-64-bit-support-now-available/

**Comment by sergiot** [comment]
Tried to compile from source but a 64bit project still can't see it as a valid module.

**Comment by etruta** [comment]
Probably because you using the old version of Parse.framework.

https://parse.com/questions/when-is-the-64bit-ios-version-coming-out

**Comment by almsx** [comment]
greetings:

I copied the latest version of the module Parse.framework h followed the steps to compile the module for 64-bit without success.

Also I changed the parameters in the xcodeproject file on standard architectures throwing this from terminal:

CompileC build / parsemodule.build / Release-iphoneos / parsemodule.build / / Objects ARMv7-Normal / Normal ParseSingleton.m ParseSingleton.o ARMv7 objective-c com.apple.compilers.llvm.clang.1_0.compiler
CompileCbuild/parsemodule.build/Release-iphoneos/parsemodule.build/Objects-normal/armv7/ComElijahwindsorParsemoduleModule.o Normal Classes / ComElijahwindsorParsemoduleModule.m ARMv7 objective-c com.apple.compilers.llvm.clang.1_0.compiler
CompileCbuild/parsemodule.build/Release-iphoneos/parsemodule.build/Objects-normal/arm64/ComElijahwindsorParsemoduleModule.o Normal Classes / ComElijahwindsorParsemoduleModule.m arm64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
CompileC build / parsemodule.build / Release-iphoneos / parsemodule.build / / Objects arm64-Normal / Normal ParseSingleton.m ParseSingleton.o arm64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(4 failures)
xcodebuild failed

**Comment by sergiot** [comment]
@etruta, seems you have probably compiled this successfully for x64? Would you mind sharing the binaries?

**Comment by Adipa** [comment]
I think i managed to compile this to 64 bit, ti-64 and xcrun lipo see it as a valid 64 bit architecture but my (3.5 GA) titanium project can't build with this in place (everyother module went to 64 bit without problems).
Uploading it in the hopes that it may work on someone elses project, hope it helps. 
Heres the link: 
Please note this is from a forked repo for this found here: https://github.com/raymondkam/ios-parse-titanium-module
Hopefully the additions in it won't affect your purposes but fair warning. 
https://www.mediafire.com/?k5j7ygmvu0sfcxu

**Comment by Adipa** [comment]
@sergiot Did you have any progress with implementing 64 bit on the module. I have got access to a copy of the module (with 64 bit support) that is known to be working, can you see if this works on your project? 
https://www.mediafire.com/?gevvrfa5ixd94gh

**Comment by roysinclair** [comment]
@Adipa  i can confirm it's not working for me

**Comment by sergiot** [comment]
@Adipa, no. I can't believe Titanium never did anything to get some official support or implemented something on their own.

I have gone full Native. This is absolutely ridiculous.

**Comment by jfencil** [comment]
This module does not work for 64 bit. When will this be updated? Yes we can rebuild the module for 64 bit and change for 64 bit architectures, but this does not effect the com.elijahwindsor.parsemodule-iphone-1.0.zip file at all! Please help asap!

**Comment by boarnoah** [comment]
@jfencil Issue is that even though we can build the module for 64 bit with Xcode the parse.framework doesn't support 64 bit. The version that DOES support 64 bit has a bunch of breaking changes(including the pfacebook stuff been removed) so its going to require updating a lot of the source from the parsemodule code to get it to work properly with 64 bit. 

EDIT: Here is the page talking about the changes required: http://blog.parse.com/learn/engineering/facebooks-latest-ios-sdk-update-and-what-it-means-for-you/

**Comment by jfencil** [comment]
Thank you for your response @Adipa and I apologize for the delayed response, but unfortunately the link that you provided is a blank page on the parse site. 

This is a very URGENT issue that needs to be fixed ASAP 

EDIT: These are some helpful links that I was able to find:

https://parse.com/questions/sdk-upgrade-on-ios
https://parse.com/questions/what-is-the-best-way-to-update-the-parse-sdk-in-my-ios-app

FROM THE LINKS ABOVE:

"I usually navigate to the project using the Finder, and delete the Parse.framework folder. Then extract the new SDK into the same folder, launch Xcode and perform a Clean, followed by Build."

THE ABOVE DOES NOT WORK FOR ME, I am presented with 19 errors, due to the pfacebook stuff being removed as you had said previously.

If anyone has any hints to a successful way to include the newest parse framework (1.7.1) for the elijahwinsor.parsemodule I would really appreciate it! Thank you!

**Comment by boarnoah** [comment]
Believe me there was a page there with instructions on what lines need updating to support the new facebook sdk stuff, really strange that it disappeared off to somewhere, I asked about it on stackoverflow just now, hope someone replies on that(http://stackoverflow.com/questions/30264415/updating-parse-for-64-bit-compatiblity).

Anyway as you found out there is a bunch of errors that pop up I attempted to fix it while following the instructions on the blog page but didn't get too far with it.

**Comment by jfencil** [comment]
Thank you very much once again @Adipa. Maybe we can work together to figure this thing out.

**Comment by boarnoah** [comment]
Kind of dont have my development enivornment right now (used this module at my previous workplace), they still havent reenabled parse for the 64 bit update yet will try to keep you informed on any progress they have with it in case that helps.

**Comment by Adipa** [comment]
Kind of dont have my development enivornment right now (used this module at my previous workplace), they still havent reenabled parse for the 64 bit update yet will try to keep you informed on any progress they have with it in case that helps.

**Comment by jfencil** [comment]
I also keep you informed if I come up with a solution. Really appreciate your help @Adipa!

**Comment by nicpeck** [comment]
Thanks for looking into this! Any progress yet?

**Comment by E2010** [comment]
I have created a 64-bit module, but only functions regarding Objects and Push Notifications implemented.   If these are the functions you need,  goto https://github.com/E2010/ios64-parse-titanium to get it.

**Comment by jfencil** [comment]
Was anyone able to implement the module above by E2010 above? ^

**Comment by E2010** [comment]
@jfencil  We already has one project using this module submit to App Store successfully (with 64bit support). Any problem when you try it?

**Comment by jfencil** [comment]
Yes, I am not quite able to get it to work at the moment. @Adipa any comments?

**Comment by kevinambruster** [comment]
I managed to get this to build by changing this line in titanium.xconfig

//TITANIUM_SDK = /Library/Application Support/Titanium/mobilesdk/osx/3.5.1.GA
TITANIUM_SDK = ~/Library/Application Support/Titanium/mobilesdk/osx/$(TITANIUM_SDK_VERSION)

from this post. https://developer.appcelerator.com/question/136061/cannot-create-an-iphone-module-since-upgrading-to-studio-20-og-sdk-201ga2

**Comment by krmannix** [comment]
@kevinambruster Awesome, thanks for that - that fixed my problem as well.

**Comment by jfencil** [comment]
@kevinambruster The default code in the elijah module is as follows:

TITANIUM_SDK = ~/Library/Application Support/Titanium/mobilesdk/osx/$(TITANIUM_SDK_VERSION)

Therefore, it would not make sense to change that line of code...? Any suggestions?
